### ISSUE TITLE
Add the new batch settings to the JavaScript driver

### ISSUE THREAD

**Issue by AtnNn** [rethinkdb/rethinkdb#3161]
```
RqlDriverError: Found maxBatchRows which is not a valid option. valid options are {useOutdated: <bool>, noreply: <bool>, timeFormat: <string>, groupFormat: <string>, binaryFormat: <string>, profile: <bool>, durability: <string>, arrayLimit: <number>}.
```

Assigning to myself.

**Comment by gchpaco** [comment]
As an accident of working on #2052 I think this is solved. :)

**Comment by gchpaco** [comment]
Solved as a consequence of the patch to #2052.
